### DISCUSSION TITLE
No route found for path 'socketcluster/'

### DISCUSSION THREAD

**Discussion by thoward27** [django/channels#1490] [General] [answered]
After completing the tutorial and integrating channels into my application, I am consistently seeing a very odd error message:

```
web_1        | 2020-08-31T19:26:25+00:00 ERROR ws_protocol.py:123 MainProcess.django-main-thread [Failure instance: Traceback: <class 'ValueError'>: No route found for path 'socketcluster/'.
web_1        | /usr/local/lib/python3.8/site-packages/autobahn/websocket/protocol.py:2839:processHandshake
web_1        | /usr/local/lib/python3.8/site-packages/txaio/tx.py:366:as_future
web_1        | /usr/local/lib/python3.8/site-packages/twisted/internet/defer.py:151:maybeDeferred
web_1        | /usr/local/lib/python3.8/site-packages/daphne/ws_protocol.py:72:onConnect
web_1        | --- <exception caught here> ---
web_1        | /usr/local/lib/python3.8/site-packages/twisted/internet/defer.py:151:maybeDeferred
web_1        | /usr/local/lib/python3.8/site-packages/daphne/server.py:200:create_application
web_1        | /usr/local/lib/python3.8/site-packages/channels/staticfiles.py:41:__call__
web_1        | /usr/local/lib/python3.8/site-packages/channels/routing.py:54:__call__
web_1        | /usr/local/lib/python3.8/site-packages/channels/sessions.py:47:__call__
web_1        | /usr/local/lib/python3.8/site-packages/channels/sessions.py:145:__call__
web_1        | /usr/local/lib/python3.8/site-packages/channels/sessions.py:169:__init__
web_1        | /usr/local/lib/python3.8/site-packages/channels/middleware.py:31:__call__
web_1        | /usr/local/lib/python3.8/site-packages/channels/routing.py:150:__call__
web_1        | ]
web_1        | 2020-08-31T19:26:30+00:00 INFO runserver.py:160 MainProcess.django-main-thread WebSocket DISCONNECT /socketcluster/ [172.18.0.1:59486]
```

I am running the webserver using `manage.py runserver` inside of a docker-compose container. 

What I would expect, given this error, is that somewhere I have `/socketcluster/` defined as a path, yet alas, I do not define such a path. 

I'm sure Channels is looking for something I haven't given it, but *what* it's looking for is a total mystery to me

**Comment by carltongibson** [comment]
Hi @thoward27 — It looks to me as if the **client** is requesting the `/socketcluster/` path — so (presuming you're using JavaScript) I'd check the JS request code to see how you can configure the request. 

Ideally you'd post that info, as there's not much to go on here.

**Reply by thoward27** [reply]
Interesting @carltongibson, I have *very* little javascript in my project and did not see these exceptions until I integrated channels, the javascript added during this process looks like this:

```
  const fooSocket = new WebSocket(
    (window.location.protocol == 'https:') ? 'wss://' : 'ws://'
      + window.location.host
      + '/ws/foo/bar/'
      + '{{ foo.pk }}/'
  );
  fooSocket.onmessage = function (e) {
    const data = JSON.parse(e.data);
    if ('log_message' in data) {
      document.querySelector('#foo-logs').value += (data.log_message + '\n');
    }
    if ('status' in data) {
      document.querySelector('#foo-status').innerHTML = data.status;
    }
  };
  fooSocket.onclose = function (e) {
    console.error('Foo socket closed unexpectedly.');
  }
```

Other than that I have Bootstrap 4.0, jQuery 3.2.1, and popper.js 1.12.9 loaded in the background.

FWIW the error crops up regardless of what page I'm on.

**Reply by carltongibson** [reply]
>  `No route found for path 'socketcluster/`

OK, but SOMETHING is requesting `/socketcluster/`...

**Reply by thoward27** [reply]
Any tips for debugging this? Figuring out where that request is coming from? I just see no reason for that to be happening

**Reply by carltongibson** [reply]
Not really, sorry... it must be in your code somewhere, or what of your dependencies... — the string `socketcluster` doesn't appear in the Channels, channels_redis or Daphne repos.

**Comment by CisarJosh** [comment]
@thoward27 Facing the same issue. So if you were able to track it down I'd appreciate any ideas. Likewise if I figure it out I'll send over my details.

**Reply by carltongibson** [reply]
Hi @CisarJosh — If you can put together a minimal reproduce project for this we can look at it...

**Reply by thoward27** [reply]
@CisarJosh I have not been able to pin it down, but the error only seems to occur when using django's `runserver`. Using Daphne seems to alleviate the issue.

**Comment by mikemanger** [comment, marked as answer]
For me there was a browser extension querying http://127.0.0.1:8000/socketcluster/. If I change the runserver port then the log message goes away.

**Reply by carltongibson** [reply]
Super. There you are. Thanks!
